mani_skill/envs/tasks/digital_twins/push_T.py

```diff
@@ -0,0 +1,509 @@
+            # we can then create a pose object using Pose.create_from_pq to then set the cube pose with. Note that even though our quaternion
+            # is not batched, Pose.create_from_pq will automatically batch p or q accordingly
+            # furthermore, notice how here we do not even using env_idx as a variable to say set the pose for objects in desired
+            # environments. This is because internally any calls to set data on the GPU buffer (e.g. set_pose, set_linear_velocity etc.)
+            # automatically are masked so that you can only set data on objects in environments that are meant to be initialized
+            obj_pose = Pose.create_from_pq(p=target_region_xyz, q=q)
+            self.tee.set_pose(obj_pose)
+
+            # here we set the location of that red/white target (the goal region). In particular here, we set the position to be in front of the cube
+            # and we further rotate 90 degrees on the y-axis to make the target object face up
+
+            xyz = torch.zeros((b, 3))
+            xyz[:] = self.ee_starting_pos2D
+            self.ee_goal_pos.set_pose(
+                Pose.create_from_pq(
+                    p=xyz,
+                    q=euler2quat(0, np.pi / 2, 0),
+                )
+            )
+
+    def evaluate(self):
+        ##success is where the overlap is over intersection thresh and ee dist to start pos is less than it's own thresh
+        inter_area = self.pseudo_render_intersection()
+        tee_place_success = (inter_area) >= self.intersection_thresh
+
+        # ee_to_start_pose = self.agent.tcp.pose.p - self.ee_starting_pos3D
+        # ee_to_start_dist = torch.linalg.norm(ee_to_start_pose, axis=1)
+        # reset_ee_success = ee_to_start_dist <= self.ee_dist_to_start_thresh
+
+        # success = (tee_place_success & reset_ee_success)
+
+        success = tee_place_success
+
+        return {
+            "success": success,
+        }
+
+    def _get_obs_extra(self, info: Dict):
+        # some useful observation info for solving the task includes the pose of the tcp (tool center point) which is the point between the
+        # grippers of the robot
+        obs = dict(
+            tcp_pose=self.agent.tcp.pose.raw_pose,
+        )
+        if self._obs_mode in ["state", "state_dict"]:
+            # if the observation mode is state/state_dict, we provide ground truth information about where the cube is.
+            # for visual observation modes one should rely on the sensed visual data to determine where the cube is
+            obs.update(
+                goal_pos=self.goal_tee.pose.p,
+                obj_pose=self.tee.pose.raw_pose,
+                #ee_restart_goal_pos=self.ee_goal_pos.pose.p ##for full implementation
+            )
+        return obs
+
+    def compute_dense_reward(self, obs: Any, action: Array, info: Dict):
+        ##reward for overlap of the tees
+
+        #legacy reward
+        #reward = self.pseudo_render_reward()
```

Comment:
leave a comment maybe about why the original diffusion policy paper reward is not used?